### PR TITLE
Add GoalQuery to PGM api

### PR DESCRIPTION
**Pull request by KingOfSquares**
Signed-off-by: KingSimon <19822231+KingOfSquares@users.noreply.github.com>

### CODE REVIEW

File: core/src/main/java/tc/oc/pgm/api/filter/query/GoalQuery.java [hunk 1]
```diff
@@ -0,0 +1,8 @@
+package tc.oc.pgm.api.filter.query;
+
+import tc.oc.pgm.goals.Goal;
+import tc.oc.pgm.goals.GoalDefinition;
+
+public interface GoalQuery extends MatchQuery {
+  Goal<? extends GoalDefinition> getGoal();
```
Comment (Pablete1234) [inline core/src/main/java/tc/oc/pgm/api/filter/query/GoalQuery.java]:
```suggestion
public interface GoalQuery<T extends GoalDefinition> extends MatchQuery {
  Goal<T> getGoal();
```

File: core/src/main/java/tc/oc/pgm/filters/query/GoalQuery.java [hunk 2]
```diff
@@ -1,14 +1,18 @@
...
 import tc.oc.pgm.goals.Goal;
+import tc.oc.pgm.goals.GoalDefinition;
 
-/**
- * Currently, the only thing this class does is derive the Match from a Goal. In the future, we may
- * have filters that respond specifically to goal queries, but currently we do not.
- */
-public class GoalQuery extends MatchQuery {
+public class GoalQuery extends MatchQuery implements tc.oc.pgm.api.filter.query.GoalQuery {
+  private final Goal<? extends GoalDefinition> goal;
```
Comment (Pablete1234) [inline core/src/main/java/tc/oc/pgm/filters/query/GoalQuery.java]:
```suggestion
public class GoalQuery<T extends GoalDefinition> extends MatchQuery implements tc.oc.pgm.api.filter.query.GoalQuery<T> {
  private final Goal<T> goal;
```

File: core/src/main/java/tc/oc/pgm/filters/query/GoalQuery.java [hunk 3]
```diff
@@ -1,14 +1,18 @@
...
 
-  public GoalQuery(Goal goal) {
+  public GoalQuery(Goal<? extends GoalDefinition> goal) {
```
Comment (Pablete1234) [inline core/src/main/java/tc/oc/pgm/filters/query/GoalQuery.java]:
```suggestion
  public GoalQuery(Goal<T> goal) {
```

File: core/src/main/java/tc/oc/pgm/filters/query/GoalQuery.java [hunk 4]
```diff
@@ -1,14 +1,18 @@
...
     super(null, goal.getMatch());
+    this.goal = goal;
+  }
+
+  @Override
+  public Goal<? extends GoalDefinition> getGoal() {
```
Comment (Pablete1234) [inline core/src/main/java/tc/oc/pgm/filters/query/GoalQuery.java]:
```suggestion
  public Goal<T> getGoal() {
```

**Review by Pablete1234 — CHANGES_REQUESTED** [review]
You should use probably be using generics as they are meant to be used, not just as wildcards everywhere

**Comment by KingOfSquares** [comment]
> You should use probably be using generics as they are meant to be used, not just as wildcards everywhere

Good point.

**Review by Pablete1234 — APPROVED** [review]

**Review by Electroid — APPROVED** [review]

**Pull request merged**